test/unit/connection_string.test.ts

```diff
@@ -832,4 +834,53 @@ describe('Connection String', function () {
         .that.matches(/useUnifiedTopology has no effect/);
     });
   });
+
+  describe('when mongodbLogPath is in options', function () {
+    const loggerFeatureFlag = Symbol.for('@@mdb.enableMongoLogger');
+
+    let stderrStub;
+    let stdoutStub;
+
+    before(() => {
```

Comment:
This should probably be `beforeEach` and `afterEach` to be safe because otherwise your mocks won't get reset properly between test cases; your current assertions are resilient against this (so even though the 1st and 3rd cases are checking the same stderr stub, they are checking it for different things), but it would be easy to add a test case into this set that is inadvertently asserting against the first call into the stub instead of the test case's own call because the author didn't realize that the describe setup was using before and after hooks instead of beforeEach and afterEach.

Reply:
I see, my reasoning for having specific logs was so the tests didn't somehow access other tests data, but I'm now realizing that's because I was using before/after instead of `beforeEach/afterEach`. I'll change the functions.